applications/FSIapplication/custom_utilities/nodal_update_utilities.h

```diff
@@ -0,0 +1,395 @@
+    /*@{ */
+
+
+    /*@} */
+    /**@name Protected Inquiry */
+    /*@{ */
+
+
+    /*@} */
+    /**@name Protected LifeCycle */
+    /*@{ */
+
+private:
+
+    /*@} */
+    /**@name Static Member Variables */
+    /*@{ */
+
+
+    /*@} */
+    /**@name Member Variables */
+    /*@{ */
+
+
+    /*@} */
+    /**@name Private Operators*/
+    /*@{ */
+
+
+    /*@} */
+    /**@name Private Operations*/
+    /*@{ */
+
+
+    /*@} */
+    /**@name Private  Access */
+    /*@{ */
+
+
+    /*@} */
+    /**@name Private Inquiry */
+    /*@{ */
+
+
+    /*@} */
+    /**@name Un accessible methods */
+    /*@{ */
+
+    /*@} */
+
+}; /* Class NodalUpdateBaseClass */
+
+
+/** Short class definition.
+Detail class definition.
+*/
+template <unsigned int TDim>
+class NodalUpdateNewmark : public NodalUpdateBaseClass<TDim>
+{
+
+public:
+
+    /** Type Definitions
+    */
+
+    /*@{ */
+
+    //~ /** Counted pointer of ClassName */
+    KRATOS_CLASS_POINTER_DEFINITION( NodalUpdateNewmark );
+    /*@} */
+
+    /** Constructor.
+     */
+
+    /**
+    * Empty constructor
+    */
+    NodalUpdateNewmark(const double BossakAlpha = -0.3)
+    {
+        mBossakAlpha = BossakAlpha;
+        mBossakBeta = 0.5 - mBossakAlpha;
+        mBossakGamma = 0.25*std::pow(1.0 - mBossakAlpha, 2.0);
+    }
+
+
+    /*@} */
+
+    /** Copy constructor.
+    */
+
+    /*@{ */
+    NodalUpdateNewmark(const NodalUpdateNewmark& Other);
+    /*@{ */
+
+    /** Destructor.
+     */
+
+    /*@{ */
+    virtual ~NodalUpdateNewmark()
+    {
+    }
+
+    /*@} */
+    /**@name Public Operators*/
+    /*@{ */
+
+    /**
+     * Computes the displacement time derivatives according to the computed displacement values using the Newmark (Bossak) scheme.
+     * @param rModelPart: modelpart in where the nodal update is to be performed
+     * @param timeStep: time step value
+     */
+    virtual void UpdateTimeDerivatives(ModelPart& rModelPart,
+                                       const double timeStep)
+    {
+
+        auto& rLocalMesh = rModelPart.GetCommunicator().LocalMesh();
+        ModelPart::NodeIterator local_mesh_nodes_begin = rLocalMesh.NodesBegin();
+        #pragma omp parallel for firstprivate(local_mesh_nodes_begin)
+        for(int k=0; k<static_cast<int>(rLocalMesh.NumberOfNodes()); ++k)
+        {
+            const ModelPart::NodeIterator it_node = local_mesh_nodes_begin+k;
+
+            const array_1d<double, 3>& umesh_n = it_node->FastGetSolutionStepValue(MESH_DISPLACEMENT, 1);  // Previous step mesh displacement
+            const array_1d<double, 3>& vmesh_n = it_node->FastGetSolutionStepValue(MESH_VELOCITY, 1);      // Previous step mesh velocity
+            const array_1d<double, 3>& amesh_n = it_node->FastGetSolutionStepValue(MESH_ACCELERATION, 1);  // Previous step mesh acceleration
+
+            const array_1d<double, 3>& umesh_n1 = it_node->FastGetSolutionStepValue(MESH_DISPLACEMENT, 1); // Current step mesh displacement
```

Comment:
this is accessing to the old step, not to the current one

Reply:
My bad. Copy & Paste error... Thank you!